src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.SpeculativeMemberSemanticModel.cs

```diff
@@ -51,6 +51,11 @@ protected override void AnalyzeBoundNodeNullability(BoundNode boundRoot, Binder
                 NullableWalker.AnalyzeWithoutRewrite(Compilation, MemberSymbol as MethodSymbol, boundRoot, binder, diagnostics, createSnapshots);
             }
 
+            protected override bool IsNullableAnalysisEnabled()
+            {
+                return _parentSemanticModelOpt.IsNullableAnalysisEnabledAtSpeculativePosition(OriginalPositionForSpeculation, Root);
```

Comment:
This is unconditionally dotting off a field marked with the `Opt` suffix. Should we at least have an `Debug.Assert` this is non-null in the ctor?